components/infobox/wikis/starcraft/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,500 @@
+local _awardAchievements = {}
+local _raceData
+
+local CustomPlayer = Class.new()
+
+local CustomInjector = Class.new(Injector)
+
+local _args
+local _player
+
+function CustomPlayer.run(frame)
+	local player = Person(frame)
+	_args = player.args
+	_player = player
+
+	player.getStatusToStore = CustomPlayer.getStatusToStore
+	player.adjustLPDB = CustomPlayer.adjustLPDB
+	player.nameDisplay = CustomPlayer.nameDisplay
+	player.calculateEarnings = CustomPlayer.calculateEarnings
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+	player.getWikiCategories = CustomPlayer.getWikiCategories
+	player.getPersonType = CustomPlayer.getPersonType
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'status' then
+		return {
+			Cell{
+				name = 'Race',
+				content = {CustomPlayer._getRaceData(_args.race, _RACE_FIELD_AS_CATEGORY_LINK)}
+			}
+		}
+	elseif id == 'role' then return {}
+	elseif id == 'region' then return {}
+	elseif
+		id == 'history' and
+		string.match(_args.retired or '', '%d%d%d%d')
+	then
+		table.insert(widgets, Cell{
+				name = 'Retired',
+				content = {_args.retired}
+			})
+	end
+	return widgets
+end
+
+function CustomInjector:addCustomCells(widgets)
+	-- switch to enable yearsActive once 1v1 matches have been converted to match2 storage
+	local yearsActive = Logic.readBool(_args.enableYearsActive)
+		and Namespace.isMain() and CustomPlayer._getMatchupData() or nil
+
+	local currentYearEarnings = _earningsGlobal[tostring(_CURRENT_YEAR)]
+	if currentYearEarnings then
+		currentYearEarnings = Math.round{currentYearEarnings}
+		currentYearEarnings = '$' .. mw.language.new('en'):formatNum(currentYearEarnings)
+	end
+
+	return {
+		Cell{
+			name = 'Approx. Winnings ' .. _CURRENT_YEAR,
+			content = {currentYearEarnings}
+		},
+		Cell{name = 'Years active', content = {yearsActive}}
+	}
+end
+
+function CustomPlayer.nameDisplay()
+	CustomPlayer._getRaceData(_args.race)
+	local raceIcon = RaceIcon.getNormalIcon{_raceData.race}
+	local name = _args.id or _player.pagename
+
+	return raceIcon .. '&nbsp;' .. name
+end
+
+function CustomPlayer._getRaceData(race, asCategory)
+	race = string.lower(race or _UNKNOWN_RACE)
+	race = CleanRace[race] or race
+	local raceTable = _RACE_DATA[race]
+
+	local faction, faction2
+	if race == _RACE_ALL_SHORT then
+		faction = _RACE_ALL
+	else
+		faction = (raceTable or {})[1]
+		faction2 = (raceTable or {})[2]
+	end
+
+	local display
+	if not raceTable and race ~= _UNKNOWN_RACE then
+		display = '[[Category:InfoboxRaceError]]<strong class="error">' ..
+			mw.text.nowiki('Error: Invalid Race') .. '</strong>'
+	else
+		if asCategory then
+			for raceIndex, raceValue in ipairs(raceTable or {}) do
+				raceTable[raceIndex] = ':Category:' .. raceValue .. ' Players|' .. raceValue .. ']]'
+					.. '[[Category:' .. raceValue .. ' Players'
+			end
+		end
+		if raceTable then
+			display = '[[' .. table.concat(raceTable, ']],&nbsp;[[') .. ']]'
+		end
```

Comment:
```suggestion
		if raceTable then
			if asCategory then
				for raceIndex, raceValue in ipairs(raceTable) do
					raceTable[raceIndex] = ':Category:' .. raceValue .. ' Players|' .. raceValue .. ']]'
						.. '[[Category:' .. raceValue .. ' Players'
				end
			display = '[[' .. table.concat(raceTable, ']],&nbsp;[[') .. ']]'
		end
```

Reply:
"end" missing in there somewhere 
will check tomorrow

Reply:
Added the missing end

Reply:
solved it a bit differently